framework/generated/vulkan_generators/vulkan_ascii_consumer_body_generator.py

```diff
@@ -137,32 +144,45 @@ def generate_feature(self):
                 cmddef += self.make_consumer_func_decl(
                     return_type, 'VulkanAsciiConsumer::Process_' + cmd, values
                 ) + '\n'
+                # Generate the code to output the return value if any:
                 return_val = ""
                 if not 'void' in return_type:
-                    return_val = ', ToString(returnValue, toStringFlags, tabCount, tabSize)'
+                    if 'VkResult' == return_type:
+                        # Dispatch to specialization for VkResult and let WriteApiCallToFile add quotes:
+                        return_val = ', ToString(returnValue)'
+                    elif 'VkDeviceAddress' == return_type or ("Get" in cmd and "Address" in cmd):
+                        # Output as hex, and let WriteApiCallToFile add quotes:
+                        return_val = ', PtrToString(returnValue)'
+                    elif 'VkBool32' == return_type:
+                        # Output as json true/false without quotes:
+                        return_val = ', Bool32ToString(returnValue), false'
+                    else:
+                        # Ints use the template ToString and are not quoted:
+                        print("Info, return type: ", return_type, " handled as default from function: ", cmd)
+                        return_val = ', ToString(returnValue), false'
+                # Generate the top of the function body:
```

Comment:
This is the core of the change: examining the type of the return value and name of the function to work out which ToString function to dispatch to and whether to put quotation marks around it in the json output (yes for pointers and enums, no for bools and ints (unless they are device addresses) for example).